pkgs/by-name/el/element-desktop/package.nix

```diff
@@ -42,29 +42,34 @@ stdenv.mkDerivation (
       hash = desktopSrcHash;
     };
 
-    offlineCache = fetchYarnDeps {
-      yarnLock = finalAttrs.src + "/yarn.lock";
-      sha256 = desktopYarnHash;
+    # TODO: fetchYarnDeps currently does not deal properly with a dependency
+    # declared as a pin to a commit in a specific git repository.
+    # While it does download everything correctly, `yarn install --offline`
+    # always wants to `git ls-remote` to the repository, ignoring the local
+    # cached tarball.
+    offlineCache = callPackage ./yarn.nix {
+      inherit (finalAttrs) version src;
+      hash = desktopYarnHash;
     };
 
     nativeBuildInputs = [
       yarn
-      fixup-yarn-lock
       nodejs
+      typescript
       makeWrapper
       jq
+      nodePackages_latest.prettier
     ] ++ lib.optionals stdenv.hostPlatform.isDarwin [ desktopToDarwinBundle ];
 
     inherit seshat;
 
     configurePhase = ''
       runHook preConfigure
 
-      export HOME=$(mktemp -d)
-      yarn config --offline set yarn-offline-mirror $offlineCache
-      fixup-yarn-lock yarn.lock
-      yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
-      patchShebangs node_modules/
+      mkdir -p node_modules/
+      cp -r $offlineCache/node_modules/* node_modules/
+      sed -i 's/matrix-gen-i18n/node node_modules\/matrix-web-i18n\/scripts\/gen-i18n.js/gi' package.json
```

Comment:
We should use substituteInPlace with --replace-fail to notice when this no longer matches.